### PR TITLE
MFLT-9619 Add Linux on the "Software & Hardware version" page of the docs

### PR DESCRIPTION
**Pull request by martijnthe**
- Added Linux subsections to the "Software & Hardware version" page.
- Added a new `memfault-device-info` reference page.
- Documented the requirements for the Hardware Version, Software Type and Software Version strings in the doc instead of referring to the (outdated) HTTP API doc.

### CODE REVIEW

File: docs/platform/software-version-hardware-version.mdx [hunk 1]
```diff
@@ -22,6 +12,13 @@ can be used to identify different device types being tracked in the same Project
 (i.e. `smart-toaster` and `smart-oven`) and different board revisions of the
 same device type (i.e. `smart-toaster-evt`, `smart-toaster-pvt`, etc.).
 
+:::tip Requirements
+
+Hardware Versions must be between 1 and 128 characters long and can only contain
+the following characters: `a-z`, `A-Z`, `0-9` and `-_.+:`.
+
+:::
```
Comment (martijnthe) [inline docs/platform/software-version-hardware-version.mdx:20]: I decided to add these inline, instead of referring the reader at the top of the page to the somewhat obscure and slightly out-of-date [API doc](https://api-docs.memfault.com/#memfault-resources).

**Review by noahp — APPROVED** [review]

File: docs/linux/reference-memfault-device-info.mdx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+### Purpose of `memfault-device-info`
+
+`memfault-device-info` is a program that `memfaultd` depends on to retrieve
+static information about the device it is running on. When `memfaultd` (or
+`memfaultctl`) runs, it will execute `memfault-device-info` and parse the output
+to collect this information.
+
+As an integrator, you will need to implement this program and make it available
+on the `PATH`.
```
Comment (sarfata) [inline docs/linux/reference-memfault-device-info.mdx:21]: A bit of a nitpick but this lefts me wondering "whose PATH?"

We mean the PATH of `memfaultd`. 

Looks like right now it is:
```
root@qemuarm64:/etc/systemd# systemctl --system show-environment
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

```suggestion
As an integrator, you will need to implement this program and make it available
on the `PATH`. (By default on Yocto for systemd units: `/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`).
```

File: docs/platform/software-version-hardware-version.mdx (hunk 1 above)
Comment (sarfata) [reply docs/platform/software-version-hardware-version.mdx:20]: While we are updating this I think we should also remove the `+` from the list because it will also have an ambiguous meaning in URL (space).
I have also added a comment to the Notion page @mrcljx started here: https://www.notion.so/memfault/Slashes-in-Resource-Identifiers-slugs-in-URL-paths-bc6171a817f54fdcb10754df72afa361?pvs=4

File: docs/platform/software-version-hardware-version.mdx [hunk 3]
```diff
@@ -67,6 +72,13 @@ The **Primary Software Type** is used to populate the Software Version field on
 the individual Device page, as well as when filtering for the device, and how it
 appears in various fleet-level charts.
 
+:::tip Requirements
+
+Software Types must be between 1 and 128 characters long and can only contain
+the following characters: `a-z`, `A-Z`, `0-9` and `-_.+:`.
```
Comment (sarfata) [inline docs/platform/software-version-hardware-version.mdx:78]:
```suggestion
the following characters: `a-z`, `A-Z`, `0-9` and `-_.:`.
```

File: docs/platform/software-version-hardware-version.mdx [hunk 4]
```diff
@@ -217,6 +234,14 @@ Some potentially useful things to consider:
 
 :::
 
+:::tip Requirements
+
+Aside of the requirements that follow from the chosen versioning scheme,
+Software Versions must be between 1 and 128 characters long and can only contain
+the following characters: `a-z`, `A-Z`, `0-9`, `-_.+:\()[]/` and spaces.
```
Comment (sarfata) [inline docs/platform/software-version-hardware-version.mdx:241]: The [notion page](https://www.notion.so/memfault/Slashes-in-Resource-Identifiers-slugs-in-URL-paths-bc6171a817f54fdcb10754df72afa361?pvs=4) says 
> tl;dr We allow / in some resource identifiers but they need to be encoded in the URL as %2F. We limit it because it is a surprising piece of our REST API.

Maybe "limiting" it means not documenting the potentially ambiguous ones here?

**Review by sarfata — APPROVED** [review]
Way to go above and beyond the call of duty on this ticket! 🎉

File: docs/platform/software-version-hardware-version.mdx (hunk 1 above)
Comment (martijnthe) [reply docs/platform/software-version-hardware-version.mdx:20]: `+` is pretty common to see in software versions: we've got ~76K software versions with a `+` in the prod database.

At this point, I don't think we can just take characters away and we'll need to ensure url encoding is applied to those identifiers.

File: docs/platform/software-version-hardware-version.mdx (hunk 3 above)
Comment (martijnthe) [reply docs/platform/software-version-hardware-version.mdx:78]: Leaving this because it has already been like that for years (also in the HTTP API doc).
